DQL (Docs):GraphQL -> DQL fixes (updating `Content-Type: application/graphql+-` references and noting back-compat, review and fix of add'l `GraphQL+-` references)

aaroncarey opened the pull request:
- Updated references to `Content-Type: application/graphql+-` -> `Content-Type: application/dql`
- Added notes about backward-compatibility with `Content-Type: application/graphql+-`
- Scrubbed for additional references to GraphQL +- and updated them as appropriate
- Minor rewording for clarity. Note: The rewording of the lead-in for fragments.md needs a close look from a technical perspective.

Note: I used the following commands from repo root to find and review all files that one would expect to be included in this PR, so I am confident that we got all references that we should update in the `master` branch:

`grep -r -l "application/graphql+-" content/*`
`grep -r -l "GraphQL+-" content/*`

The only item that isn't in-scope is a doc that will be deprecated in a separate PR (content/howto/migrate-dgraph-1-1.md) 

Fixes GRAPHQL-775

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/6858)
<!-- Reviewable:end -->
 
 
<!-- Dgraph:start -->
Docs Preview: [<img src="https://bl.ocks.org/prashant-shahi/raw/3a9f99bec84231cfe3c0e82cf883f159/0e588d908ad8c8b10958b87ebdd2ba68779ccf4f/dgraph.svg" height="34" align="absmiddle" alt="Dgraph Preview"/>](https://dgraph-175a43e91b-106966.surge.sh)
<!-- Dgraph:end -->

verneleem commented on wiki/content/query-language/fragments.md:
```diff
@@ -6,10 +6,14 @@ weight = 25
 +++
 
-`fragment` keyword allows you to define new fragments that can be referenced in a query, as per [GraphQL specification](https://facebook.github.io/graphql/#sec-Language.Fragments). The point is that if there are multiple parts which query the same set of fields, you can define a fragment and refer to it multiple times instead. Fragments can be nested inside fragments, but no cycles are allowed. H ...
+The `fragment` keyword lets you to define new fragments that can be referenced
```
>The `fragment` keyword lets you to define

I think there is an extra word here

> The `fragment` keyword lets you define

aaroncarey replied:
Fixed, good catch 👍

verneleem commented on wiki/content/query-language/fragments.md:
```diff
@@ -6,10 +6,14 @@ weight = 25
+in a query, per the [GraphQL specification](https://facebook.github.io/graphql/#sec-Language.Fragments).
+Fragments allow for the reuse of common repeated selections of fields, reducing
+duplicated text in the DQL documents. Fragments can be nested inside fragments,
+but no cycles are allowed in such cases. For example:
```
I would advise possible the term looping for clarity of what a cycle is being referred to:

>but no looping cycles are allowed

aaroncarey replied:
Thank you, very useful. This was the sentence that I was unsure about.

verneleem reviewed (commented):
Just my thoughts on the matter. LGTM from my technical knowledge

vardhanapoorv reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6858#-:-MLRDVdQAmu-_jMcAfqd:bg7igkv)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @bucanero, @danielmai, @JatinDevDG, @MichaelJCompton, and @NamanJain8)



<!-- Sent from Reviewable.io -->

JatinDevDG reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6858)* status: all files reviewed, 1 unresolved discussion (waiting on @aaroncarey, @bucanero, @danielmai, @MichaelJCompton, and @NamanJain8)

___
*[wiki/content/graphql/custom/graphqlpm.md, line 13 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6858#-MLROsD117wX3557sPar:-MLROsD2AY9i43W_VNwd:b-irn9zx) ([raw file](https://github.com/dgraph-io/dgraph/blob/175a43e91b612b5fc45721b034051f936e6b0c85/wiki/content/graphql/custom/graphqlpm.md#L13)):*
> ```Markdown
> API will execute that for you.
> 
> DQL lets you build custom logic that goes beyond whta is possible with the
> ```

whta??


<!-- Sent from Reviewable.io -->

aaroncarey reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6858)* status: all files reviewed, 1 unresolved discussion (waiting on @bucanero, @danielmai, @JatinDevDG, @MichaelJCompton, and @NamanJain8)

___
*[wiki/content/graphql/custom/graphqlpm.md, line 13 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6858#-MLROsD117wX3557sPar:-MLRcEvO5cNLUaMiF1mM:be7gre6) ([raw file](https://github.com/dgraph-io/dgraph/blob/175a43e91b612b5fc45721b034051f936e6b0c85/wiki/content/graphql/custom/graphqlpm.md#L13)):*
<details><summary><i>Previously, JatinDevDG (Jatin Dev) wrote…</i></summary><blockquote>

whta??
</blockquote></details>

Fixed, thanks for catching this typo. I'll find a spellcheck plugin to avoid this in the future.


<!-- Sent from Reviewable.io -->

aaroncarey commented:
Ugh... I merged after addressing comments, but before pushing to the origin branch. I opened https://github.com/dgraph-io/dgraph/pull/6859 to close this loop.
